### DISCUSSION TITLE
How do I customize - Login? ie. https://THEDOMAIN:5443/#/pages/login

### DISCUSSION THREAD

**Discussion by AppThisThing** [ant-media/Ant-Media-Server#5311] [Q&A]
Hello Ant Media,

Re: Enterprise Edition

1. How do we customized this login page?

2. Where/What is the path to the location to this file on the server to edit the text & Image?

Thanks

https://THEDOMAIN:5443/#/pages/login
<img width="1023" alt="5443:#:pages:login" src="https://github.com/ant-media/Ant-Media-Server/assets/107368149/313ed9d1-1430-48c3-aec4-ec074549b33d">

**Comment by tremzo** [comment]
The web panel is an Angular application and is open source. Its located in this Github repository https://github.com/ant-media/Ant-Media-Management-Console. To change the login page, you'll need to clone the project and run the Angular application locally to develop the web panel. The login page is located at

`src/app/pages/login/login.component.html`

Here is the page in Github https://github.com/ant-media/Ant-Media-Management-Console/blob/master/src/app/pages/login/login.component.html. 

Then you'll need to build the project and upload the static files to the `root` application located at:

`​/usr/local/antmedia/webapps/root`

​Make sure not to remove the `WEB-INF` and ​`META-INF` folders.

**Reply by AppThisThing** [reply]
Thanks for the information.

Would you have an video tutorial for this?
How to Clone then Build the project, something for newbies ?

**Reply by mekya** [reply]
What about doing this together @AppThisThing, @timantmedia  in a meeting and record the session?
After that, @NurC123 can make the video ready to publish on youtube

**Reply by tremzo** [reply]
Ok this sounds like a good idea. I'll do that.

**Reply by nhatvuong2504** [reply]
I have found documentation for the customization using Ant-Media-Management-Console-master but there are no specific instructions on how to install nor instructions, Can someone help me

**Comment by AppThisThing** [comment]
Thank you so much for creating a video tutorial on cloning and building projects for newbies! Your contribution in providing guidance and insights into this technology is highly valuable. I am confident that your video will serve as a valuable resource for individuals interested in learning about Ant Media Server. Keep up the excellent work! @NurC123  @mekya @timantmedia

**Reply by mekya** [reply]
Thank you @AppThisThing 

Why don't you join us and have a coffee together as mentioned in about us page?
https://antmedia.io/about-us/

Did you have the session @timantmedia?

**Reply by tremzo** [reply]
@mekya I was waiting for some clarity on other aspects of the management console, however, @AppThisThing if you want to go over just modifying the login page we can do that together in a meeting. Feel free to schedule it here https://calendly.com/tim-bobker

**Reply by mekya** [reply]
I can also join the session and we can have remote coffee together  :)

**Reply by tremzo** [reply]
@nhatvuong2504 @AppThisThing Apologies for the delay, but I made a short video tutorial on how to do some basic customization of the Ant Media Server web panel here https://www.youtube.com/watch?v=eXUocOTJlFE&t=26s

**Comment by AppThisThing** [comment]
Hello All,

Sounds good! Please add me to the schedule.

**Reply by tremzo** [reply]
@AppThisThing Please schedule at your convenience here then https://calendly.com/tim-bobker

**Reply by tremzo** [reply]
@mekya customer would like to also add some custom form fields in the login form. @AppThisThing can you please let us know what additional form fields you want to add prior to the meeting so I can review it for you? 

You mention a possible payment feature (option), so are you looking for payment integration with login? Can you provide some details?

**Reply by AppThisThing** [reply]
The main goal is to learn how to customize the the login page.
A how to for newbies. 

This is only if possible.

**Change:** Logo
**Change:** Background Image
**Add:** Custom Form Fields
**User:** Sign Registration / Login 
**Social Login:** Google, Meta, Linked, Instagram 
**Payment Option:** Coinbase, Paypal, Stripts, & others - during sign up / Registration --- >> the user would select a Subscription $5, $10, $20

**Reply by tremzo** [reply]
@AppThisThing Subscriptions and additional registration fields are advanced topics. The API has all the required functions to register users and authenticate for example, to register a new user you can use https://antmedia.io/rest/#/ManagementRestService/addUser so building a custom registration page is possible outside of the Ant Media Server application/web panel.

However, to implement a subscription system with payments, would require custom development of Ant Media Server which is out of the scope of just customising the login page and is not available by default. The same for social login, these would need integration into those social login platforms which are not available by default. 

These are topics that have not yet been covered, I can add these to our blog post publishing queue so in the future we can publish how this can be done. 

I can still go ahead with the basic customisation of the login page which requires cloning the management console Github repository, making the edits and then building the static files to upload to the root application.

**Reply by tremzo** [reply]
Further to the meeting yesterday here are some useful links:

- Introduction to plugin architecture https://www.youtube.com/watch?v=hiEr3Y4ya7Q and https://antmedia.io/plugins-will-make-ant-media-server-more-powerful/
- Building an Ant Media Application https://www.youtube.com/watch?v=ZTevUpoYcjw and https://antmedia.io/how-to-create-custom-new-applications/

@mekya One question that was asked which I cannot advise, is how can you add new fields to the registration / add user form in the web panel and can this be exposed as a public signup/registration form? If the user wants to add a subscription model to Ant Media server, would modifying the web panel be a recommended way or building out a completely separate application that interacts with Ant Media Server using the API / SDK?

Let us know if you have any outstanding questions.

**Comment by nhatvuong2504** [comment]
I need instructions on how to install Ant-Media-Management-Console on my server to change the logo and translate the language accordingly. I don't know how to do it? Please guide me step by step

**Reply by rahul78275** [reply]
Hello @nhatvuong2504 

Clone the repository from [here](https://github.com/ant-media/Ant-Media-Management-Console)

Make Sure you have node installed (Ver : 14.8.0), if not then install from [here](https://nodejs.org/en)

Install NPM
`npm install -g @angular/cli@10.0.5`

Once you are done with these installation, verify the installation using (node -v and npm -v) command.
Navigate to the project path on cli which you cloned

Use command 
`npm install `// This will install all the dependencies. It will take few minutes and might give some warning so ignore them.

Once your dependency is downloaded.

Use command
`npm start`

This will start the project on you local machine at (http://localhost:4200)

Now in order to customize the login page navigate to this path (src/app/pages/login/login.component.html), Make the changes of your choice and save the file. 

Browser will automatically update the changes done by you.

Once every thing is fine. Create the final build which you will upload on the server.

To create the build use command

`ng build --prod`

After the build creation you will get a **dist** folder in your project directory.

Now you need to keep a backup of ​(/usr/local/antmedia/webapps/root) root directory (For safety purpose).

After taking the backup delete all the files from root directory and upload all the files of dist folder to the root directory of server.

If you face any issue during the process, let me know.

Thank You.

**Reply by nhatvuong2504** [reply]
I tried but can't access localhost:4200?

I have installed nodejs V14.8.0 npm v6.14.7
I installed it successfully: npm install -g @angular/cli@10.0.5

I extracted Ant-Media-Management-Console according to the link below:
`​/usr/local/antmedia/webapps/root/​ant-media/Ant-Media-Management-Console`

`npm install`
`npm start`

But can't access: http://localhost:4200/

**Reply by rahul78275** [reply]
Can we connect on a meet call ?

**Reply by nhatvuong2504** [reply]
I'm not good at English, if you can show me how to record the screen?
Or explain to me which step I did wrong?

**Reply by rahul78275** [reply]
Me too not very good in english 😄 
It would be difficult to explain here, this thread will go longer. 
Lets try https://meet.google.com/ywt-zxww-eez

**Reply by ankurjain23** [reply]
I have a question Can we access the page using the HTTP://IP address:4200?

**Reply by yashtandon113** [reply]
@ankurjain23 
Yes, it is doable.

Please check the following discussion to change the HTTP/S port.
https://github.com/orgs/ant-media/discussions/5533

Please restart the server after making the changes.

**Reply by ankurjain23** [reply]
The localhost page is opening but while opening with the system IP address is not working.. any solution for this

**Reply by rahul78275** [reply]
Hi @ankurjain23 

If you are working on local system then try to run 

`ng serve --host 0.0.0.0` 

Now you can access you application like : http://192.168.XXX.XXX:4200

this [link ](https://stackoverflow.com/questions/43492354/how-to-allow-access-outside-localhost)may be helpful for you

**Reply by ankurjain23** [reply]
HI @rahul78275 It is showing Command 'ng' not found, but can be installed with sudo apt install ng-common
can you pls make me this understand

**Reply by rahul78275** [reply]
Hi @ankurjain23 

Can we quick connect [here ](https://meet.google.com/ybq-ieor-pvc)

**Reply by ankurjain23** [reply]
sure

**Reply by rahul78275** [reply]
Possibly missing angular cli package. Try ( sudo npm install -g @angular/cli)

Linking of npm with cli  Try (sudo npm link @angular/cli)

**Reply by ankurjain23** [reply]
Hi Rahul, can we change the footer of the web login page?
<img width="756" alt="image" src="https://github.com/ant-media/Ant-Media-Server/assets/85350913/bbbf28f3-83eb-4ee3-8046-fc3eb370d8c0">

**Reply by rahul78275** [reply]
Hi @ankurjain23 
I will recommend to have this line if not an issue. 
However you can make changes in this file "src/app/shared/footer/footer.component.html".

**Reply by ankurjain23** [reply]
Thanks @rahul78275

**Comment by rahul78275** [comment]
Hi @USAMAWIZARD 

Dependency installation sometimes creates issue as i discussed with you earlier.
me and @nhatvuong2504 tried to fix his issue on digital ocean, but no success due to error while creating the build. 

I had installed the same version of node as mentioned in travis file. During the build process we got error message. 

```
ERROR in ./node_modules/jwa/index.js
Module not found: Error: Can't resolve 'crypto' in '/node_modules/jwa'

ERROR in ./node_modules/jws/lib/verify-stream.js
Module not found: Error: Can't resolve 'stream' in '/root/Ant-Media-Management-Console/node_modules/jws/lib'

ERROR in ./node_modules/jws/lib/sign-stream.js
Module not found: Error: Can't resolve 'stream' in '/root/Ant-Media-Management-Console/node_modules/jws/lib'

ERROR in ./node_modules/jws/lib/data-stream.js
Module not found: Error: Can't resolve 'stream' in '/root/Ant-Media-Management-Console/node_modules/jws/lib'
```
I installed stream package manually, this fixed the last 3 error, but crypto still having issue.

Tried npm install after deleting package-lock.json but no luck.

However @nhatvuong2504  only had to change the logo, So  manually replacing the logo image did the trick for him.

**Comment by AppThisThing** [comment]
Please include me on any live video tutorial again. I only took quick hand notes.
I was given a "**_GREAT LESSON_**", but failed to video record it.
I never was able to get it working on my live server,  I tried to implement but just failed.

I'm on Digital Ocean's Server - Ubuntu 20.04 (LTS) x64

One more time so I can record, and master this!

Thanks

**Comment by yashtandon113** [comment]
@rahul78275 @nhatvuong2504 
Please try the below solution if you have not tried this yet. It worked for me last time when I faced the same issue.
https://github.com/auth0/node-jsonwebtoken/issues/471#issuecomment-398798497

Please let me know if that helps.

**Comment by AppThisThing** [comment]
Awesome video!

I wasn't doing it on the "*Local*" machine.
I was trying to perform this on the live Ubuntu server.

Any more "Enterprise Edition" video lessons would be great.

Your help is and was much appreciated!

AppThisThing!
David & Melanie

**Comment by ankurjain23** [comment]
Hi Rahul,

My laptop suddenly crashed and not working properly.. will catch you again
after resolve my laptop issue.


BR,
Ankur
